agasc/supplement/utils.py

```diff
@@ -73,3 +79,60 @@ def get_supplement_table(name, agasc_dir=None, as_dict=False):
         out = Table(dat)
 
     return out
+
+
+def _load_or_create(filename, table_name):
+    try:
+        table = Table.read(filename, format='hdf5', path=table_name)
+    except OSError as e:
+        logger.debug(f'Creating agasc supplement table "{table_name}"')
+        if not filename.exists() or str(e) == f'Path {table_name} does not exist':
+            table = Table()
+        else:
+            raise
+    return table
+
+
+def save_version(filename, **kwargs):
+    """Save the version of a supplement table to the "versions" table.
+
+    Along with the version, the time of update is also added to another table called "last_updated"
+
+    Example usage::
+
+        from agasc.supplement.utils import save_version
+        save_version('agasc_supplement.h5', mags='4.10.2')
+
+    The different versions can be retrieved from the default supplement::
+
+        from agasc.supplement.utils import get_supplement_table
+        versions = get_supplement_table('versions')
+
+    :param filename: pathlib.Path
+    :param kwargs: dict
+        A dictionary of table name/version pairs,
+    """
+    import agasc
+    version_dtype = '<U32'
+    filename = Path(filename)
+
+    versions = _load_or_create(filename, 'versions')
```

Comment:
Can we name this table `agasc_versions`? I got quite confused because I think of `versions` as being versions of the tables, not versions of the package that wrote the tables.

Reply:
I changed the table name. We could also add the table version itself, but this is another case where scope is creeping up. The next obvious question is whether we want to track table versions and guarantee back compatibility with older table versions. If we wanted to do that, then perhaps the supplement would have been implemented differently from the ground up.